mne/channels/tests/test_montage.py

```diff
@@ -1293,4 +1354,115 @@ def test_transform_to_head_and_compute_dev_head_t():
         ))
 
 
+def test_set_montage_with_mismatching_ch_names():
+    """Test setting a DigMontage with mismatching ch_names."""
+    raw = read_raw_fif(fif_fname)
+    montage = make_standard_montage('mgh60')
+
+    # 'EEG 001' and 'EEG001' wont match
+    # with pytest.raises(RuntimeError, match='channel .*not found'):
+    with pytest.warns(RuntimeWarning, match='not set 60 channel positions'):
+        raw.set_montage(montage, raise_if_subset=False)
+
+    montage.ch_names = [  # modify the names in place
+        name.replace('EEG', 'EEG ') for name in montage.ch_names
+    ]
+    raw.set_montage(montage)  # does not raise
+
+
+def test_set_montage_with_sub_super_set_of_chnames():
+    """Test info and montage ch_names matching criteria."""
+    N_CHANNELS = len('abcdef')
+    montage = _make_toy_dig_montage(N_CHANNELS, coord_frame='head')
+
+    # Montage and info match
+    _ = create_info(
+        ch_names=list('abcdef'), sfreq=1, ch_types='eeg', montage=montage
+    )
+
+    # Montage is a SUPER-set of info
+    with pytest.warns(RuntimeWarning, match='super-set'):
```

Comment:
superset and subset are written without a `-`

https://en.wikipedia.org/wiki/Subset